common/log/platform/boost/boost_log.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <cassert>
+#include <iostream>
+#include <fstream>
+
+#include <boost/smart_ptr/shared_ptr.hpp>
+#include <boost/core/null_deleter.hpp>
+#include <boost/date_time/posix_time/posix_time.hpp>
+#include <boost/log/common.hpp>
+#include <boost/log/expressions.hpp>
+#include <boost/log/attributes.hpp>
+#include <boost/log/sinks.hpp>
+#include <boost/log/sources/logger.hpp>
+#include <boost/log/utility/manipulators/add_value.hpp>
+#include <boost/log/attributes/scoped_attribute.hpp>
+#include <boost/log/support/date_time.hpp>
+
+#include <common/log.hpp>
+
+namespace mender::log {
+
+namespace logging = boost::log;
+namespace expr = boost::log::expressions;
+namespace sinks = boost::log::sinks;
+namespace attrs = boost::log::attributes;
+namespace src = boost::log::sources;
+namespace keywords = boost::log::keywords;
+
+using namespace std;
+
+
+static void logfmt_formatter(logging::record_view const &rec, logging::formatting_ostream &strm) {
+	strm << "record_id=" << logging::extract<unsigned int>("RecordID", rec) << " ";
+
+	auto level = logging::extract<LogLevel>("Severity", rec);
+	if (level) {
+		strm << "severity=" << level << " ";
+	}
+
+	auto val = logging::extract<boost::posix_time::ptime>("TimeStamp", rec);
+	if (val) {
+		strm << "time=\"" << val.get() << "\" ";
+	}
+
+	auto name = logging::extract<std::string>("Name", rec);
+	if (name) {
+		strm << "name=\"" << name.get() << "\" ";
+	}
+
+	auto fields = logging::extract<std::vector<LogField>>("ExtraFields", rec);
+	if (fields) {
+		for (auto const &field : fields.get()) {
+			strm << field.key << "=" << field.value << " ";
+		}
+	}
+
+	for (auto f : rec.attribute_values()) {
+		auto field = logging::extract<LogField>(f.first.string(), rec);
+		if (field) {
+			strm << field.get().key << "=\"" << field.get().value << "\" ";
+		}
+	}
+
+	strm << "msg=\"" << rec[expr::smessage] << "\" ";
+}
+
+
+static boost::shared_ptr<boost::log::v2s_mt_posix::sinks::synchronous_sink<
```

Comment:
I tried to change it initially, but it keeps spewing type errors...

Reply:
Hmm, ok, if you don't find a solution, please tell me and I'll try to have a look. I'd really like us to avoid having two different pointer types if possible. It will just be a mess, and will grow the binary size unnecessarily as well.